pkg/discovery/dns/provider.go

```diff
@@ -125,6 +127,7 @@ func (p *Provider) Resolve(ctx context.Context, addrs []string) {
 			level.Error(p.logger).Log("msg", "dns resolution failed", "addr", addr, "err", err)
 			// Use cached values.
 			p.RLock()
+			errorList = append(errorList, err)
```

Comment:
I think this change can be moved outside of Rlock.

Reply:
I was thinking of putting it outside the lock, the consistency of the error in the list does not matter. Thanks for the suggestion!